### PR TITLE
Implement MSC2815: allow room moderators to view redacted event content

### PR DESCRIPTION
**Pull request by tulir**
This pull request implements the query parameter described in MSC2815 to allow room moderators and server admins to get the original content of redacted events using the `/rooms/{roomId}/event/{eventId}` endpoint.

Signed-off-by: Tulir Asokan <tulir@maunium.net>

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog).
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: synapse/rest/client/room.py [hunk 1]
```diff
@@ -643,18 +645,55 @@ def __init__(self, hs: "HomeServer"):
...
+        ):
+            power_level_event = await self._state.get_current_state(
+                room_id, EventTypes.PowerLevels, ""
+            )
+
+            auth_events = {}
+            if power_level_event:
+                auth_events[(EventTypes.PowerLevels, "")] = power_level_event
+
+            redact_level = event_auth.get_named_level(auth_events, "redact", 50)
```
Comment (squahtx) [inline synapse/rest/client/room.py:680]: I'm surprised we don't have a series of constants somewhere for "default-\<action\>-power-level"

**Review by squahtx — CHANGES_REQUESTED** [review]
Looks good, just a few minor comments and questions

File: synapse/storage/databases/main/events_worker.py [hunk 2]
```diff
@@ -303,6 +303,15 @@ async def get_received_ts(self, event_id: str) -> Optional[int]:
...
         )
 
+    async def have_censored_event(self, event_id: str) -> Optional[bool]:
+        return await self.db_pool.simple_select_one_onecol(
+            table="redactions",
+            keyvalues={"redacts": event_id},
+            retcol="have_censored",
+            desc="get_have_censored",
+            allow_none=True,
+        )
```
[the remark this answers is not in the document]
Comment (tulir) [reply synapse/storage/databases/main/events_worker.py]: Indeed there can, there's an index on `redacts`, but I didn't notice that it wasn't a unique index.

I'm not sure if it there are cases where some `have_censored`s are false and some are true, but I changed the function to fetch all of them and return `any(values)` (so true if any are true, false if empty/all false)

**Review by squahtx — APPROVED** [review]
LGTM! Thanks for making the changes!

**Pull request merged**